pkg/allowdenylist/allowdenylist.go

```diff
@@ -18,13 +18,23 @@ package allowdenylist
 
 import (
 	"errors"
-	"regexp"
 	"strings"
+	"time"
+
+	regexp "github.com/dlclark/regexp2"
+	"k8s.io/klog/v2"
 
 	generator "k8s.io/kube-state-metrics/v2/pkg/metric_generator"
 )
 
-// AllowDenyList encapsulates the logic needed to filter based on a string.
+// Use ECMAScript as the default regexp spec to support lookarounds (#2594).
+var regexpDefaultSpec regexp.RegexOptions = regexp.ECMAScript
+
+func init() {
```

Comment:
Can we not make use of an init() function here and instead wrap the regexp.Compile into a function that sets this?

Reply:
I think in general init() is a bad pattern that we should avoid introducing because of side effects, import order etc.
Can we add this to the `New()` func instead?
https://github.com/kubernetes/kube-state-metrics/pull/2616/files#diff-b14b1102ef04d884e648b90166b5d35ee63edac2a544ef311770f5f289df48f4L36

Reply:
Done, PTAL.